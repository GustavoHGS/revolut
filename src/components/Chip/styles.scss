@import 'assets/_variables.scss';

.container {
  height: fit-content;
  background-color: $blue-dark;
  font-family: GloberSemiBold;
  font-size: 13px;
  padding: 6px 8px 4px 8px;
  border-radius: 12px;
  color: $white;
  display: flex;
  flex-direction: row;
  align-items: center;
  line-height: 1;
  width: fit-content;
}

.primary {
  background-color: $blue;
}