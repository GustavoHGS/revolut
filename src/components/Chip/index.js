/* eslint react/forbid-prop-types: off */
import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import './styles.scss'

const Chip = ({
  label, primary, content, style,
}) => (
  <div
    className={classNames('container', primary ? 'primary' : '')}
    style={style}
  >
    {label || content()}
  </div>
)

export default Chip

Chip.propTypes = {
  label: PropTypes.string,
  primary: PropTypes.bool,
  content: PropTypes.func,
  style: PropTypes.object,
}

Chip.defaultProps = {
  label: '',
  primary: false,
  content: () => {},
  style: null,
}
