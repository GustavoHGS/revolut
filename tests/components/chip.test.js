import React from 'react'
import renderer from 'react-test-renderer'
import Chip from '../../src/components/Chip'

describe('Input Component', () => {
  it ('matches the chip snapshot', () => {
    const nodeTree = renderer.create(<Chip />).toJSON()
    expect(nodeTree).toMatchSnapshot()
  })
})
