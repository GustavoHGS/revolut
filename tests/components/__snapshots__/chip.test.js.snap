// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component matches the chip snapshot 1`] = `
<div
  className="container"
  style={null}
/>
`;
